amundsen_application/static/js/components/common/ResourceList/index.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import Pagination from 'react-js-pagination';
+import ResourceListItem from 'components/common/ResourceListItem';
+import { Resource } from 'interfaces';
+import { ITEMS_PER_PAGE, PAGINATION_PAGE_RANGE } from './constants';
+
+
+export interface ResourceListProps {
+  items: Resource[];
+  source: string;
+
+  // The following props are only used if 'paginate' is set to true
+  paginate?: boolean;
+  activePage?: number;
+  isFullList?: boolean;
+  itemsCount?: number;
+  itemsPerPage?: number;
+  onPagination?: (pageNumber: number) => void;
+}
+
+interface ResourceListState {
+  activePage: number;
+}
+
+class ResourceList extends React.Component<ResourceListProps, ResourceListState> {
+  public static defaultProps: Partial<ResourceListProps> = {
+    paginate: false,
+    activePage: 0,
+    itemsPerPage: ITEMS_PER_PAGE,
+    isFullList: true,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { activePage: this.props.activePage };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.activePage !== prevProps.activePage) {
+      this.setState({ activePage: this.props.activePage });
+    }
+  }
+
+  onPagination = (rawPageNum: number) => {
+    const activePage = rawPageNum - 1;
+    if (this.props.onPagination !== undefined) {
+      this.props.onPagination(activePage);
+    } else {
+      this.setState({ activePage });
```

Comment:
Does this not cause another `setState` top trigger in `componentDidUpdate`?

Reply:
It does, but the condition on 39 stops it from doing anything. This is another weird thing were the `activePage` can either be managed internally inside this component or externally a la `SearchPage`. This is another reason I considered splitting this into multiple components, but that ended up being pretty messy as well.

Reply:
The check on line 39 compares the current props to previous props, given that on this line we update the state and not the props, how does the check stop `setState` from firing again?

Reply:
So there are two pathways for pagination and `activePage`.

1. The `activePage` is managed internally only (MyBookmarks, PopularTables). `onPagination` triggers `setState` on line 49. The check on line 39 doesn't run because `this.props.activePage` should be `undefined`.
2. `activePage` is managed externally (SearchPage). `onPagination` fires `this.props.onPagination` which updates the `activePage` on the external component. This in turn should update `this.prop.activePage` and trigger the `setState` in `componentDidUpdate`


I agree this behavior is unclear and convoluted. I'll try to see if we can simplify this somehow. Ideally I would like all page management to be done internally.